api/filter/network/tcp_proxy.proto

```diff
@@ -30,53 +30,59 @@ message TcpProxy {
   // Configuration for access logs.
   repeated AccessLog access_log = 5;
 
-  // A TCP proxy route consists of a set of optional L4 criteria and the
-  // name of a cluster. If a downstream connection matches all the
-  // specified criteria, the cluster in the route is used for the
-  // corresponding upstream connection. Routes are tried in the order
-  // specified until a match is found. If no match is found, the connection
-  // is closed. A route with no criteria is valid and always produces a
-  // match.
-  message TCPRoute {
-    // The cluster to connect to when a the downstream network connection
-    // matches the specified criteria.
-    string cluster = 1;
+  message DeprecatedV1 {
+    // [V2-API-DIFF] This is deprecated in v2. Routes will be matched using
+    // the FilterChainMatch in Listeners.
+    //
+    // A TCP proxy route consists of a set of optional L4 criteria and the
+    // name of a cluster. If a downstream connection matches all the
+    // specified criteria, the cluster in the route is used for the
+    // corresponding upstream connection. Routes are tried in the order
+    // specified until a match is found. If no match is found, the connection
+    // is closed. A route with no criteria is valid and always produces a
+    // match.
+    message TCPRoute {
+      // The cluster to connect to when a the downstream network connection
+      // matches the specified criteria.
+      string cluster = 1;
```

Comment:
We don't need stuff that already appears in the outer proto, e.g. there is `cluster` above.

Reply:
I think we do need this as we need to be able to route to different clusters? How does the v1 JSON look? I think it's possible we left the outer cluster for back-compat. I don't think we need to do that here and could possibly require a catch-all route for this in v2.